ori/volume/server/volume.go

```diff
@@ -16,235 +16,127 @@ package server
-	corev1.ClaimBound:   ori.VolumeState_VOLUME_AVAILABLE,
-	corev1.ClaimLost:    ori.VolumeState_VOLUME_ERROR,
-}
-
-func generateWWN(wwnPrefix string) (string, error) {
-	// prefix is optional, set to 1100AA for private identifier
-	wwn := wwnPrefix
+const (
+	VolumeClassLabel  = "volume-class"
+	SnapshotNameLabel = "snapshot-name"
+)
 
-	// use UUIDv4, because this will generate good random string
-	wwnUUID, err := uuid.NewRandom()
+func (s *Server) convertImageToOriVolume(ctx context.Context, log logr.Logger, image *api.Image) (*ori.Volume, error) {
+	metadata, err := s.getOriObjectMetadata(&image.Metadata)
 	if err != nil {
-		return "", fmt.Errorf("failed to generate UUIDv4 for WWN: %w", err)
-	}
-
-	// append hex string without "-"
-	wwn += strings.Replace(wwnUUID.String(), "-", "", -1)
-
-	// WWN is 64Bit number as hex, so only the first 16 chars are returned
-	return wwn[:16], nil
-}
-
-func (s *Server) convertOnmetalVolumeState(state corev1.PersistentVolumeClaimPhase) (ori.VolumeState, error) {
-	if state, ok := onmetalVolumeStateToORIState[state]; ok {
-		return state, nil
-	}
-	return 0, fmt.Errorf("unknown onmetal volume state %q", state)
-}
-
-func (s *Server) convertOnmetalVolumeAccess(ctx context.Context, volume *AggregateCephVolume) (*ori.VolumeAccess, error) {
-	if volume.Pvc.Status.Phase != corev1.ClaimBound {
-		return nil, nil
+		return nil, fmt.Errorf("error getting ori metadata: %w", err)
 	}
 
-	pv := &corev1.PersistentVolume{}
-	if err := s.client.Get(ctx, types.NamespacedName{Name: volume.Pvc.Spec.VolumeName, Namespace: volume.Pvc.Namespace}, pv); err != nil {
-		return nil, client.IgnoreNotFound(fmt.Errorf("unable to get pv: %w", err))
-	}
-	if pv.Status.Phase != corev1.VolumeBound {
-		return nil, nil
-	}
-
-	monitors, err := s.getMonitorList(ctx)
+	spec, err := s.getOriVolumeSpec(image)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get monitor list for volume: %w", err)
+		return nil, fmt.Errorf("error getting ori resources: %w", err)
 	}
 
-	imageKey, err := s.getImageKeyFromPV(pv)
+	state, err := s.getOriState(image.Status.State)
 	if err != nil {
-		return nil, fmt.Errorf("failed to provide image name: %w", err)
+		return nil, fmt.Errorf("error getting ori state: %w", err)
 	}
 
-	wwn, err := generateWWN(s.wwnPrefix)
-	if err != nil {
-		return nil, fmt.Errorf("unable to generate wwn: %w", err)
+	var access *ori.VolumeAccess
+	if state == ori.VolumeState_VOLUME_AVAILABLE {
+		access, err = s.getOriVolumeAccess(ctx, log, image)
+		if err != nil {
+			return nil, fmt.Errorf("error getting ori volume access: %w", err)
+		}
 	}
 
-	credentials, err := s.getVolumeAccessCredentials(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("unable to get volume access credentials: %w", err)
+	return &ori.Volume{
+		Metadata: metadata,
+		Spec:     spec,
+		Status: &ori.VolumeStatus{
+			State:  state,
+			Access: access,
+		},
+	}, nil
+}
+
+func wwnFromImageID(id string) string {
```

Comment:
With this, you're making implicit assumptions on the generated name of the image below. What if the name length becomes shorter with a new release? Panic?
The wwn generation has to work regardless of the image name.